service/src/main/java/com/ge/predix/acs/config/CommonRedisConfig.java

```diff
@@ -0,0 +1,64 @@
+package com.ge.predix.acs.config;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.data.redis.serializer.StringRedisSerializer;
+
+@Configuration
+@Profile({ "cloud-redis", "redis" })
+public class CommonRedisConfig {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CommonRedisConfig.class);
+
+    private RedisConnectionFactory decisionRedisConnectionFactory;
+    private RedisConnectionFactory resourceRedisConnectionFactory;
+    private RedisConnectionFactory subjectRedisConnectionFactory;
+
+    @Autowired
+    public CommonRedisConfig(final RedisConnectionFactory decisionRedisConnectionFactory,
+            final RedisConnectionFactory resourceRedisConnectionFactory,
+            final RedisConnectionFactory subjectRedisConnectionFactory) {
+        this.decisionRedisConnectionFactory = decisionRedisConnectionFactory;
+        this.resourceRedisConnectionFactory = resourceRedisConnectionFactory;
+        this.subjectRedisConnectionFactory = subjectRedisConnectionFactory;
+    }
+
+    @Bean(name = {"redisTemplate", "decisionCacheRedisTemplate"})
```

Comment:
This is required. Redis auto configuration needs a bean named redisTemplate or it will try and create one for us.